components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs

```diff
@@ -565,8 +565,9 @@ internal sealed class SubscriptionWrapper : ISubscription {
                     .GetValueOrDefault(session.DefaultSubscription.KeepAliveCount);
                 _subscription.MonitoredItems?.ForEach(m => {
                     if (m.HeartbeatInterval != null && m.HeartbeatInterval != TimeSpan.Zero) {
-                        var itemKeepAliveCount = (uint)m.HeartbeatInterval.Value.TotalMilliseconds /
-                            (uint)_subscription.Configuration.PublishingInterval.Value.TotalMilliseconds;
+                        var _publishingInterval = (uint)(_subscription.Configuration.PublishingInterval == TimeSpan.Zero? 
+                                                  TimeSpan.FromSeconds(1) : _subscription.Configuration.PublishingInterval).Value.TotalMilliseconds;
+                        var itemKeepAliveCount = (uint)m.HeartbeatInterval.Value.TotalMilliseconds / _publishingInterval;
                         revisedKeepAliveCount = GreatCommonDivisor(revisedKeepAliveCount, itemKeepAliveCount);
```

Comment:
looking at the way how keepAlive is dependent on heartbeat interval, different heartbeat interval should not be mixed in same subscriptions unless they are multiples of each other, otherwise keep alive can become very small.

Reply:
Maybe we open an issue for this scenario and discuss it there.